models/quality_measures/intermediate/cqm438_statin_therapy_for_patients_with_cardiovascular_disease/quality_measures__int_cqm438_numerator.sql

```diff
@@ -0,0 +1,185 @@
+{{ config(
+     enabled = var('quality_measures_enabled',var('claims_enabled',var('clinical_enabled',var('tuva_marts_enabled',False)))) | as_bool
+   )
+}}
+
+with denominator as (
+
+    select
+          patient_id
+        , performance_period_begin
+        , performance_period_end
+        , measure_id
+        , measure_name
+        , measure_version
+    from {{ ref('quality_measures__int_cqm438_denominator') }}
+
+)
+
+, statin_codes as (
+
+    select
+          code
+        , code_system
+        , concept_name
+    from {{ ref('quality_measures__value_sets') }}
+    where lower(concept_name) in  (
+          'high intensity statin therapy'
+        , 'low intensity statin therapy'
+        , 'moderate intensity statin therapy'
+        , 'statin therapy'
+    )
+
+)
+
+, procedures as (
+
+    select
+        patient_id
+      , procedure_date
+      , coalesce (
+              normalized_code_type
+            , case
+                when lower(source_code_type) = 'cpt' then 'hcpcs'
+                when lower(source_code_type) = 'snomed' then 'snomed-ct'
+                else lower(source_code_type)
+              end
+          ) as code_type
+        , coalesce (
+              normalized_code
+            , source_code
+          ) as code
+    from {{ ref('quality_measures__stg_core__procedure') }}
+
+)
+
+, procedure_statin_related as (
+
+    select 
+          procedures.patient_id
+        , procedures.procedure_date as evidence_date
+    from procedures
+    inner join statin_codes
+        on procedures.code = statin_codes.code
+            and procedures.code_type = statin_codes.code_system
+
+)
+
+, pharmacy_claims_statin_related as (
+
+    select
+        patient_id
+      , dispensing_date as evidence_date
+      , ndc_code
+    from {{ ref('quality_measures__stg_pharmacy_claim') }} as pharmacy_claims
+    inner join statin_codes
+    on pharmacy_claims.ndc_code = statin_codes.code
+        and lower(statin_codes.code_system) = 'ndc'
+
+)
+
+, medication_statin_related as (
+
+    select
+          patient_id
+        , coalesce(dispensing_date, prescribing_date) as evidence_date
+        , source_code
+        , source_code_type
+    from {{ ref('quality_measures__stg_core__medication') }} as medications
+    inner join statin_codes
+        on medications.source_code = statin_codes.code
+        and medications.source_code_type = statin_codes.code_system
+        
+)
+
+, condition_statin_related as (
+
+    select 
+          conditions.patient_id
+        , conditions.recorded_date as evidence_date
+        , conditions.source_code
+        , conditions.source_code_type
+    from {{ ref('quality_measures__stg_core__condition') }} conditions
+    inner join statin_codes
+        on conditions.source_code_type = statin_codes.code_system
+            and conditions.source_code = statin_codes.code
+
+)
```

Comment:
I think we can remove this. There are no code systems related to conditions for these concepts.

Reply:
Thank you pointing this out @sarah-tuva. Just out of curiosity, what kind of code systems do we generally see in conditions? One I believe is `snomed-ct`. Can all procedure code be in conditions code as well?

Reply:
@utsavpaudel we usually see the following code systems for conditions (aka diagnoses).

* snomed-ct
* icd-10-cm
* icd-9-cm

The code system snomed-ct is unique in that it contains codes for conditions, procedures, and observations. It is very common in clinical data sources that come from an EMR (electronic medical record).